### PR TITLE
Tickets/pipe2d-734: Handle broken red shutter in ISR

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**

### CODE REVIEW

File: python/lsst/obs/pfs/isrTask.py [hunk 1]
```diff
@@ -107,37 +135,92 @@ def runDataRef(self, sensorRef):
...
+                            dataId["visit"] = dataId0["visit"] + dv + 10000
+
+                            md = sensorRef.get("raw_md")
+                            if md.get('DATA-TYP', "").upper() == "BIAS":
+                                self.log.info("Reading visit %d for broken red shutter correction",
+                                              sensorRef.dataId["visit"])
+
+                                exp2 = self.runDataRef(sensorRef).exposure
+                        except NoResults as e:
+                            print("RHL", e)
```
Comment (hassanxp) [inline python/lsst/obs/pfs/isrTask.py:210]: Remove debug statement

File: python/lsst/obs/pfs/isrTask.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        camera = sensorRef.get("camera")
+        if camera is None and self.config.doAddDistortionModel:
+            raise RuntimeError("config.doAddDistortionModel is True "
+                               "but could not get a camera from the butler")
+        isrData = self.readIsrData(sensorRef, ccdExposure)
+
+        result = self.run(ccdExposure, camera=camera, **isrData.getDict())
+
+        if self.config.doBrokenRedShutter and sensorRef.dataId["arm"] == 'r':
+            if self.config.doBrokenRedShutter and self.config.brokenRedShutter.useAnalytic:
```
Comment (hassanxp) [inline python/lsst/obs/pfs/isrTask.py:139]: Would be good to check and raise an error if ``doBrokenRedShutter`` is ``True`` and the arm is not 'r'

**Review by hassanxp — APPROVED** [review]
This is fine. Minor comments added.

File: python/lsst/obs/pfs/isrTask.py (hunk 2 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/pfs/isrTask.py:139]: Actually the `+ 10000` is part of this debug code, so I need to remove that too (and the format is wrong -- clearly
I added this check-the-exception code but got distracted before doing the checking).  I'm glad you spotted this.

File: python/lsst/obs/pfs/isrTask.py (hunk 1 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/pfs/isrTask.py:210]: Actually the + 10000 is part of this debug code, so I need to remove that too (and the format is wrong -- clearly
I added this check-the-exception code but got distracted before doing the checking). I'm glad you spotted this.

File: python/lsst/obs/pfs/isrTask.py (hunk 2 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/pfs/isrTask.py:139]: I don't think so, as that's a config for the processing and we're generally processing both b and r

**Pull request merged**